master/usage/ipv6.md

```diff
@@ -1,31 +1,158 @@
+	`--service-cluster-ip-range`.
+  - The `--node-ip` flag should be set to the IPv6 address of the node.
+- For kube-proxy:
+  - The `--bind-address` flag should be set to the appropriate IPv6
+    address or `::` for all IPv6 addresses on the host.
+  - The `--master` flag should be set with the IPv6 address where the
+    kube-apiserver can be accessed.
+  - The `--cluster-cidr` flag must be set to match the {{site.prodname}}
+    IPv6 IPPool.
+
+### Limitations
+
+- Currently Kubernetes supports only one IP stack version at a time. This
+  means that if you configure Kubernetes for IPv6 then {{site.prodname}}
+  should be configured to only assign IPv6 addresses.
+- You should not switch a deployment that is using IPv4 to IPv6, or
+  vise-versa but to instead setup a new cluster.
+
+### Details of Enabling IPv6
+
+To enable IPv6 operation of a {{site.prodname}} deployment follow the
+steps below.
+
+1. Download the Calico manifest with the calico-config ConfigMap and
+   calico-node Daemonset as `calico.yaml`.
+1. Update the `calico.yaml` file with the following changes:
+   - If the ipam section of the `cni_network_config` has
+     `"type": "calico-ipam"` then it should be modified to have
+     [IPv4 assigning disabled and IPv6 assigning
+     enabled](/{{page.version}}/reference/cni-plugin/configuration#ipam).
+   - Add the following environment variables to the calico-node Daemonset
+
+     ```
+     - name: CALICO_IPV6POOL_CIDR
+       value: "fd20::0/112"
+     - name: IP6
+       value: "autodetect"
+     ```
+
+     - Set the `CALICO_IPV6POOL_CIDR` environment variable to the desired pool,
+       it should match the `--cluster-cidr` passed to the
+       kube-controller-manager and to kube-proxy.
+   - Ensure the environment variable `FELIX_IPV6SUPPORT` is set `true`
+     on the calico-node Daemonset.
+1. Apply the `calico.yaml` manifest.
+
+### DNS changes
+
+It will probably be necessary to modify your DNS pod for IPv6. If you are using
+[kube-dns](/{{page.version}}/getting-started/kubernetes/installation/manifests/kubedns.yaml), 
+then the following changes will ensure IPv6 operation.
+
+- Update the image versions to at least `1.14.8`.
+- Ensure the clusterIP for the DNS service matches the one specified to
+  the kubelet as `--cluster-dns`.
+- Add `--dns-bind-address=[::]` to the arguments for the kubedns container.
+- Add `--no-negcache` to the arguments for the dnsmasq container.
+- Switch the arguments on the sidecar container from
+  ```
+  --probe=kubedns,127.0.0.1:10053,kubernetes.default.svc.cluster.local,5,A
+  --probe=dnsmasq,127.0.0.1:53,kubernetes.default.svc.cluster.local,5,A
+  ```
+  to
+  ```
+  --probe=kubedns,127.0.0.1:10053,kubernetes.default.svc.cluster.local,5,SRV
+  --probe=dnsmasq,127.0.0.1:53,kubernetes.default.svc.cluster.local,5,SRV
+  ```
+
+## Enabling IPv6 with OpenStack
+
+### Prerequisites
 
 When using {{site.prodname}} with a VM platform (e.g. OpenStack), obtaining IPv6
```

Comment:
Always a comma before and after i.e. and e.g. In this case, since the e.g. is in parentheses, you just need one after it.